app/figures.py

```diff
@@ -547,33 +576,42 @@ def create_waffle_chart(
     if not all(isinstance(item, int) for item in counts):
         raise TypeError("Counts must be integers")
 
+    # Proportional representation
+    if squares:
+        counts_pr = sainte_lague_algorithm(counts, squares)
+    else:
+        counts_pr = counts
```

Comment:
Merely a question of preference, so feel free to ignore:
```suggestion
    counts_pr = sainte_lague_algorithm(counts, squares) if squares else counts
```